sdk/search/azure-search/src/main/java/com/azure/search/models/IndexBatch.java

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.search.models;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.search.Document;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Contains a batch of document write actions to send to the index.
+ */
+@Fluent
+public class IndexBatch<T> extends IndexBatchImpl<T> {
+    /**
+     * Constructor
+     */
+    public IndexBatch() {
+        this.actions(new ArrayList<>());
+    }
+
+    /**
+     * Set the actions property: The actions in the batch.
+     *
+     * @param actions the actions value to set.
+     * @return the IndexBatch object itself.
+     */
+    public IndexBatch<T> actions(List<IndexAction<T>> actions) {
+        return (IndexBatch<T>) super.setActions(actions);
+    }
+
+
+    /**
+     * Adds an Upload IndexAction to the IndexAction chain for a document.
+     *
+     * @param documents The documents to be uploaded.
+     * @return IndexBatch with the desired actions added.
+     */
+    @SuppressWarnings("unchecked")
+    public IndexBatch<T> addUploadAction(T... documents) {
```

Comment:
@brjohnstmsft I beleive the names were discussed, but can be revised if needed. 
What would be a better naming for the methods:
 - addUploadAction(T... documents)
 - addDeleteAction(T... documents)
 - addMergeAction(T... documents)

Reply:
@navalev I was thinking `addUploadActions`, `addDeleteActions`, and `addMergeActions` since they add multiple actions. We can make this change after the merge and consulting the API review board; I was just curious if you remembered the history behind this choice.